File: java/src/com/vittee/thainum/provider/BaseNameProvider.java
package com.vittee.thainum.provider;

public class BaseNameProvider implements NameProvider {

    private final String[] names;

    public BaseNameProvider(String... names) {
        this.names = names;
    }

    @Override
    public String getName(int i) {
        return names[i];
    }
}

File: csharp/ThaiNum/ThaiNum.Library/Provider/BaseNameProvider.cs
using System;

namespace ThaiNum.Library.Provider
{
    public class BaseNameProvider : INameProvider
    {
        private readonly String[] _names;

        public BaseNameProvider(params String[] names)
        {
            this._names = names;
        }

        public String this[int i] {
            get
            {
                return _names[i];
            }
        }
    }
}
